android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/HomeListChildView.kt

```diff
@@ -118,83 +133,190 @@ fun HomeListChildView(
+						horizontalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.xs)),
+					) {
+						item {
+							ButtonWithIcon(
+								modifier = Modifier
+									.width(dimensionResource(id = R.dimen.xl))
+									.height(dimensionResource(id = R.dimen.l)),
+								iconId = R.drawable.tune_round_black_24,
+								onClick = {
+									filterType.value = FilterType.ALL
+									coroutineScope.launch {
+										isBottomSheetOpen = true
+
+									}
+								},
+								contentDescription = stringResource(id = R.string.filter_menu),
 							)
-						},
-						selectedValue = uiState.locationRange.stringId,
-						width = dimensionResource(id = R.dimen.xxxxl),
-					)
-					FilterDropDown(
-						filterName = stringResource(id = R.string.price),
-						dropDownItems = PriceRange.values().map { it.stringId }
-							.toTypedArray(),
-						updateState = { newVal ->
-							viewModel.priceRangeFilterStream.onNext(
-								PriceRange.fromInt(newVal)
+						}
+						item {
+							FilterButton(
+								filterName = stringResource(id = R.string.location),
+								onClick = {
+									filterType.value = FilterType.LOCATION
+									coroutineScope.launch {
+										isBottomSheetOpen = true
+
+									}
+								},
+
+								)
+						}
+						item {
+							FilterButton(
+								filterName = stringResource(id = R.string.price),
+								onClick = {
+									filterType.value = FilterType.PRICE
+									coroutineScope.launch {
+										isBottomSheetOpen = true
+									}
+								}
 							)
-						},
-						selectedValue = uiState.priceRange.stringId,
-						width = dimensionResource(id = R.dimen.xxxl),
+						}
+						item {
+							FilterButton(
+								filterName = stringResource(id = R.string.rooms)
+							) {
+								filterType.value = FilterType.ROOMS
+								coroutineScope.launch {
+									isBottomSheetOpen = true
+								}
+							}
+						}
+						item {
+							FilterButton(
+								filterName = stringResource(id = R.string.property_type)
+							) {
+								filterType.value = FilterType.PROPERTY_TYPE
+								coroutineScope.launch {
+									isBottomSheetOpen = true
+								}
+							}
+						}
+						item {
+							FilterButton(
+								filterName = stringResource(id = R.string.roommate),
+								onClick = {
+									filterType.value = FilterType.ROOMMATE
+									coroutineScope.launch {
+										isBottomSheetOpen = true
+									}
+								}
+							)
+						}
+					}
+				}
+				item {
+					if (isBottomSheetOpen) {
+						ModalBottomSheet(
+							modifier = Modifier.wrapContentHeight(),
+							onDismissRequest = { isBottomSheetOpen = false },
+							sheetState = modelSheetState,
+							containerColor = Color.White,
+							content = {
+								when (filterType.value) {
+									FilterType.LOCATION -> {
+										LocationFilterForm(
+											uiState.locationRange,
+											::updateLocationFilter
+										) { closeBottomSheet() }
```

Comment:
Just a small thing but you can do `::closeBottomSheet` instead of creating another lambda function